### PR TITLE
Feature/#505 api for adding and querying hub accounts

### PR DESCRIPTION
**Pull request by bothadeon**

### CODE REVIEW

File: seeds/hubParticipant.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+ * Georgi Georgiev <georgi.georgiev@modusbox.com>
+ --------------
+ ******/
+
+'use strict'
+
+const hubParticipant =
+  {
+    'participantId': 1,
```
[the remark this answers is not in the document]
Comment (bothadeon) [reply seeds/hubParticipant.js]: The Hub participant aka hub operator must always have participantId of 1, the others are just DFSP's

File: config/default.json [hunk 2]
```diff
@@ -31,6 +31,10 @@
     }
   },
   "PARTICIPANT_INITIAL_POSTITION": 0,
+  "HUB_OPERATOR": {
+    "ACCOUNTS": ["HUB_RECONCILIATION","HUB_FEE"],
```
[the remark this answers is not in the document]
Comment (bothadeon) [reply config/default.json]: Hub operator accounts moved to enums.
Hub operator id aka participantId remains in default.json
All tests passed.

File: src/admin/participants/handler.js [hunk 3]
```diff
@@ -85,6 +86,48 @@ const create = async function (request, h) {
   }
 }
 
+const account = async function (request, h) {
```
[the remark this answers is not in the document]
Comment (bothadeon) [reply src/admin/participants/handler.js]: changed handler from account to participantAccount
refer commit :

File: src/models/participant/facade.js [hunk 4]
```diff
@@ -456,7 +456,44 @@ const getParticipantLimitsByParticipantId = async (participantId, type, ledgerAc
...
+        }
+        result = await knex('participantCurrency').transacting(trx).insert(participantCurrency)
+        participantCurrency.participantCurrencyId = result[0]
+        let participantPosition = {
+          participantCurrencyId: participantCurrency.participantCurrencyId,
+          value: 0,
+          reservedValue: 0
+        }
+        result = await knex('participantPosition').transacting(trx).insert(participantPosition)
+        participantPosition.participantPositionId = result[0]
```
[the remark this answers is not in the document]
Comment (bothadeon) [reply src/models/participant/facade.js]: Console.log to the screen
===================
tmanRuntime/7.1.1","accept":"*/*","host":"localhost:3001","accept-encoding":"gzip, deflate","content-length":"48","connection":"keep-alive","traceid":"ec30676a-8842-4a45-850b-da5c06286613"}
 !**!!! Result 0 = 17**
 !!!! Result 1 = undefined
 !!!! Result 2 = undefined
2018-10-24T11:34:16.343Z - info: L1p-Trace-Id=ec30676a-8842-4a45-850b-da5c06286613 - Response: {"name":"Hub","id":"http://localhost:3001/participants/Hub","created":"2018-10-23T12:14:08.000Z","isActive":1,"links":{"self":"http:

Db.participantPosition
================
image



Conclusion:
=========
Yip result[0] in this test returns the particpantPositionId of the newly created position record.

File: src/admin/participants/handler.js [hunk 5]
```diff
@@ -85,6 +86,48 @@ const create = async function (request, h) {
...
+      if (!ledgerAccountType) {
+        throw new Errors.LedgerAccountTypeNotFoundError()
+      }
+      let accountParams = {
+        participantId: participant.participantId,
+        currencyId: request.payload.currency,
+        ledgerAccountTypeId: ledgerAccountType.ledgerAccountTypeId,
+        isActive: 1
+      }
+      let accountExists = await Participant.getParticipantAccount(accountParams)
```
[the remark this answers is not in the document]
Comment (bothadeon) [reply src/admin/participants/handler.js]: Renamed to participant Account

File: src/models/ledgerAccountType/ledgerAccountType.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ ******/
+
+'use strict'
+
+const Db = require('../../db')
+
+exports.getByName = async (accountParams) => {
+  try {
+    const participantCurrency = await Db.participantCurrency.findOne(accountParams)
+    return participantCurrency
```
[the remark this answers is not in the document]
Comment (bothadeon) [reply src/models/ledgerAccountType/ledgerAccountType.js]: Method moved.
ref 81d32cf

File: src/admin/participants/routes.js [hunk 7]
```diff
@@ -205,6 +205,29 @@ module.exports = [
       }
     }
   },
+  {
+    method: 'POST',
+    path: '/participants/{name}/accounts',
+    handler: Handler.account,
```
[the remark this answers is not in the document]
Comment (bothadeon) [reply src/admin/participants/routes.js]: Changed to participantAccount

File: src/admin/participants/handler.js [hunk 8]
```diff
@@ -219,6 +262,7 @@ const getAccounts = async function (request, h) {
 
 module.exports = {
   create,
+  account,
```
[the remark this answers is not in the document]
Comment (bothadeon) [reply src/admin/participants/handler.js]: Changed to particiPantAccount

File: src/admin/participants/routes.js [hunk 9]
```diff
@@ -205,6 +205,29 @@ module.exports = [
...
+    options: {
+      id: 'participants_accounts_create',
+      tags: tags,
+      description: 'Create new participant account',
```
[the remark this answers is not in the document]
Comment (bothadeon) [reply src/admin/participants/routes.js]: Changed to : description: 'Create new Participant and Hub operator accounts'

File: config/default.json [hunk 10]
```diff
@@ -8,7 +8,7 @@
     "SCALE": 2
   },
   "SIDECAR": {
-    "DISABLED": false,
+    "DISABLED": true,
```
[the remark this answers is not in the document]
Comment (bothadeon) [reply config/default.json]: SIDECAR set to unreal

**Comment by elnyry** [comment]
approved as per Valentin's comments

**Pull request merged**